### ISSUE TITLE
Regression when updating to node: ^23 (no-unpublished-require)

### ISSUE THREAD

**Issue by scagood** [eslint-community/eslint-plugin-n#369]
We have a regression in node: ^23

Specifically this test is failing https://github.com/eslint-community/eslint-plugin-n/blob/bf34ca53864e059e3fbf632f33429ba10a75ee9b/tests/lib/rules/no-unpublished-require.js#L201-L204

---

Example action:

https://github.com/eslint-community/eslint-plugin-n/actions/runs/11448831559/job/31853177596?pr=368

```js
  4) no-unpublished-require
       valid
         require('./');:

      AssertionError [ERR_ASSERTION]: Should have no errors but had 1: [
  {
    ruleId: 'rule-to-test/no-unpublished-require',
    severity: 1,
    message: '"./" is not published.',
    line: 1,
    column: 9,
    nodeType: 'Literal',
    messageId: 'notPublished',
    endLine: 1,
    endColumn: 13
  }
]

1 !== 0

      + expected - actual

      -1
      +0
      
      at testValidTemplate (node_modules/eslint/lib/rule-tester/rule-tester.js:925:20)
      at Context.<anonymous> (node_modules/eslint/lib/rule-tester/rule-tester.js:1274:37)
      at process.processImmediate (node:internal/timers:511:21)
```
